### PR TITLE
 #2005, #2050 - Modified tool to transfer gdm and interpretation from user/affiliation to user/affiliation.  Added missing gdm objects to be transferred.

### PR DESCRIPTION
**Pull request by gcheung-SF**

### CODE REVIEW

**Comment by gcheung-SF** [comment]
Made following changes after meeting with curators:
1. Change url to /transfer-curation and page title to "Transfer Curation".
2. Rename "Affiliation Id" to "New Affiliation Id" and move it to be last field.
3. Add new dropdown select to choose if using individual user(s) or affiliation(s) for the transfer.
4. The contributors field can contain either user uuid(s) or affiliation id(s) accordingly.
5. Add following new validation checking.
     If selected GDM/interpretation belongs to an user, then only "Individual" contributor can be selected for the transfer.  Same rule for curation belongs to an affiliation, only "Affiliation" contributor can be selected.
     If selected "Individual" contributor(s) is not the submitter of the GDM/Interpretation or  selected "Affiliation" contributor(s) does not own the GDM/Interpretation, cannot transfer the curation.
     If selected "Affiliation" contributor is the same as "New Affiliation Id", will not transfer.
     
To test:
Create new GDM with some evidences (different users/affiliations), then goto /transfer-curation page to test.
Create new interpretation with some articles/criteria (different users/affiliations), then goto /transfer-curation page to test.

File: src/clincoded/static/components/transfer_curation.js [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+        const contributorType = this.getFormValue('contributor_type')
+        const contributorUuids = this.getFormValue('contributor_uuid');
+        var valid = false;
+
+        if (contributorType === 'affiliation' && contributorUuids === affiliationId) {
+            this.setState({errorMsg: 'Selected Affiliation Id for transfer should not be the same as new Affiliation Id.'});
+        } else if (contributorType === 'individual' && object.affiliation) {
+            this.setState({errorMsg: 'Cannot transfer curation from an affiliation by selected individual user(s).'});
+        } else if (contributorType === 'individual' && !object.affiliation && contributorUuids.indexOf(object.submitted_by.uuid) === -1) {
+            this.setState({errorMsg: 'Cannot transfer curation because it is not belonged to any selected individual user(s).'});
```
[the remark this answers is not in the document]
Comment (gcheung-SF) [reply src/clincoded/static/components/transfer_curation.js]: Fixed

File: src/clincoded/static/components/transfer_curation.js [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+        } else if (contributorType === 'affiliation' && !object.affiliation) {
+            this.setState({errorMsg: 'Cannot transfer curation from an user by selected affiliation(s).'});
+        } else if (contributorType === 'affiliation' && object.affiliation && contributorUuids.indexOf(object.affiliation) === -1) {
+            this.setState({errorMsg: 'Cannot transfer curation because it is not belonged to any selected affiliation(s).'});
```
[the remark this answers is not in the document]
Comment (gcheung-SF) [reply src/clincoded/static/components/transfer_curation.js]: Fixed

**Comment by gcheung-SF** [comment]
@markmandell, @h-tong, I have to add back the existing /add-affiliation page in order to rename it to /transfer-curation page.  Since the page has already been added to the production database, the new page cannot be added.  @bryanwulf or I will manually rename the page after R31 goes live.  Thanks.

**Comment by markmandell** [comment]
> @markmandell, @h-tong, I have to add back the existing /add-affiliation page in order to rename it to /transfer-curation page. Since the page has already been added to the production database, the new page cannot be added. @bryanwulf or I will manually rename the page after R31 goes live. Please do final code review. Thanks.

Sounds good! Thanks for the update 👍

**Pull request merged**